### ISSUE TITLE
Deconvolution2D is not outputing the same shape as Documentation

### ISSUE THREAD

**Issue by louisng5** [fchollet/keras#4601]
Hi and thanks in advance,

I am using theano GPU as backend and here is my code:

`import os`
`os.environ['KERAS_BACKEND'] = 'theano'`
`from keras.layers import  Deconvolution2D`
`from keras.models import Sequential`
`import numpy as np`

`model = Sequential()`
`model.add(Deconvolution2D(3, 3, 3, output_shape=(None, 3, 14, 14), border_mode='valid', input_shape=(3, 12, 12)))`
`dummy_input = np.ones((32, 3, 12, 12))`
`preds = model.predict(dummy_input)`
`print(preds.shape)`

I got an output with shape (32, 5, 14, 3) which I cannot make sense of. Also, it is different from the documentation [https://keras.io/layers/convolutional/#deconvolution2d](url)

I am confused.

**Comment by louisng5** [comment]
Problem kind of solved after adding
`import keras`
`keras.backend.set_image_dim_ordering('th')`

I expected dim ordering is default to 'th'.

**Comment by erilyth** [comment]
@louisng5 Default ordering is 'tf'
From the documentation,
"dim_ordering: 'th' or 'tf'. In 'th' mode, the channels dimension (the depth) is at index 1, in 'tf' mode is it at index 3. It defaults to the image_dim_ordering value found in your Keras config file at ~/.keras/keras.json. If you never set it, then it will be "tf"."
